### PR TITLE
feat: follow new api convention

### PR DESCRIPTION
**Pull request by abhinav-harness**

### CODE REVIEW

File: scm/driver/harness/repo.go [hunk 1]
```diff
@@ -45,8 +53,10 @@ func (s *repositoryService) FindPerms(ctx context.Context, repo string) (*scm.Pe
 }
 
 func (s *repositoryService) List(ctx context.Context, opts scm.ListOptions) ([]*scm.Repository, *scm.Response, error) {
-	harnessURI := buildHarnessURI(s.client.account, s.client.organization, s.client.project, "")
-	path := fmt.Sprintf("api/v1/spaces/%s/repos?sort=path&order=asc&%s", harnessURI, encodeListOptions(opts))
+	queryParams := fmt.Sprintf("projectIdentifier=%s&orgIdentifier=%s&accountIdentifier=%s&routingId=%s",
+		s.client.project, s.client.organization, s.client.account, s.client.account)
+
+	path := fmt.Sprintf("api/v1/repos?%s&sort=path&order=asc&%s", queryParams, encodeListOptions(opts))
```
Comment (johannesHarness) [inline scm/driver/harness/repo.go:59]: **minor** for other lists we put queryparams at the end of the parameters.

File: scm/driver/harness/util.go [hunk 2]
```diff
@@ -32,6 +33,30 @@ func buildHarnessURI(account, organization, project, repo string) (uri string) {
 	return repo
 }
 
+func getRepoAndQueryParams(slug string) (string, string, error) {
+	params := url.Values{}
+	s := strings.TrimSuffix(slug, "/+")
+	splitSlug := strings.Split(s, "/")
+	params.Set("accountIdentifier", splitSlug[0])
+	params.Set("routingId", splitSlug[0])
```
Comment (johannesHarness) [inline scm/driver/harness/util.go:41]: we don't need `routingId` if we set `accountIdentifier`, right? or is it just to be sure?

File: scm/driver/harness/util.go [hunk 3]
```diff
@@ -32,6 +33,30 @@ func buildHarnessURI(account, organization, project, repo string) (uri string) {
 	return repo
 }
 
+func getRepoAndQueryParams(slug string) (string, string, error) {
+	params := url.Values{}
+	s := strings.TrimSuffix(slug, "/+")
+	splitSlug := strings.Split(s, "/")
+	params.Set("accountIdentifier", splitSlug[0])
```
Comment (johannesHarness) [inline scm/driver/harness/util.go:40]: to be safe, add a check for `len(splitSlug)==0`?

File: scm/driver/harness/util.go [hunk 4]
```diff
@@ -32,6 +33,30 @@ func buildHarnessURI(account, organization, project, repo string) (uri string) {
...
+	params.Set("routingId", splitSlug[0])
+	var repoId string
+	switch len(splitSlug) {
+	case 2:
+		repoId = splitSlug[1]
+	case 3:
+		params.Set("orgIdentifier", splitSlug[1])
```
Comment (johannesHarness) [inline scm/driver/harness/util.go:47]: **minor** make these constatns

File: scm/driver/harness/util.go [hunk 5]
```diff
@@ -32,6 +33,30 @@ func buildHarnessURI(account, organization, project, repo string) (uri string) {
...
+		repoId = splitSlug[2]
+	case 4:
+		params.Set("orgIdentifier", splitSlug[1])
+		params.Set("projectIdentifier", splitSlug[2])
+		repoId = splitSlug[3]
+	default:
+		return "", "", errors.New(fmt.Sprintf("split length more than %d encountered", len(splitSlug)))
```
Comment (johannesHarness) [inline scm/driver/harness/util.go:54]: you can use `fmt.Errorf(...` instead

**Review by johannesHarness — APPROVED** [review]

File: scm/driver/harness/util.go (hunk 2 above)
Comment (abhinav-harness) [reply scm/driver/harness/util.go:41]: just to be sure.

**Review by johannesHarness — APPROVED** [review]

**Pull request merged**